layers/cmd/ingest_csv.go

```diff
@@ -30,6 +31,93 @@ import (
 	"github.com/cloudprivacylabs/opencypher/graph"
 )
 
+type CSVIngester struct {
+	BaseIngestParams
+	StartRow     int
+	EndRow       int
+	HeaderRow    int
+	ID           string
+	InitialGraph string
+}
+
+func (ci CSVIngester) Run(pipeline *PipelineContext) error {
+	// pipeline.Input = strings.NewReader(context.Background().Value("filename").(string))
+	initialGraph := ci.InitialGraph
+	layer, err := LoadSchemaFromFileOrRepo(pipeline.Context, ci.CompiledSchema, ci.Repo, ci.Schema, ci.Type, ci.Bundle)
+	if err != nil {
+		return err
+	}
+
+	reader := csv.NewReader(pipeline.Input)
+	startRow := ci.StartRow
+	endRow := ci.EndRow
+	headerRow := ci.HeaderRow
+	if headerRow >= startRow {
+		return errors.New("Header row is ahead of start row")
+
+	}
+	var grph graph.Graph
```

Comment:
instead of initializing graph, use pipelineContex.graph here. See comments for pipeline context.